### PR TITLE
feat(Listing Related product and  seller information)

### PR DESCRIPTION
**Pull request by Sophie20202**
**What does this PR do?**

This is Listing related product specifically for buyers.

**Description of Task to be completed?**

This task will require A product Id in order to retrieve all its related products

**How should this be manually tested?**

This will be tested by using Get a single product with  Id

**What are the relevant pivotal tracker/Trello stories?**
id of the project
#187505367

### CODE REVIEW

**Review by fab-ryan — CHANGES_REQUESTED** [review]
why did you remove all products
- related product must be based on single not all products
- your have remove the current code which was about search and filter product
- the related product must be under sngle product

**Comment by Sophie20202** [comment]
Already resolved

File: src/controllers/productController.ts [hunk 1]
```diff
@@ -314,3 +314,83 @@ export const updateProductAvailability = async (
...
+        },
+      ],
+      limit: 4,
+    });
+
+    const relatedProductsInCategory = await excOperation<ProductAttributes[]>(
+      'Product',
+      'findAll',
+      {
+        where: {
```
Comment (fab-ryan) [inline src/controllers/productController.ts]: response {
product,
relatedproduct,
seller info
}

**Review by fab-ryan — CHANGES_REQUESTED** [review]
seller info not include but remain don't pass out the password

File: src/controllers/productController.ts [hunk 2]
```diff
@@ -219,13 +219,65 @@ export const getAProduct = async (req: Request, res: Response) => {
...
+          attributes: ['id', 'name', 'phone', 'email'],
+        },
+        {
+          model: Database.Category,
+          as: 'category',
+          attributes: ['id', 'name'],
+        },
+      ],
+      limit: 4,
+    });
```
Comment (fab-ryan) [inline src/controllers/productController.ts]: here you will return also this product meaning not related

File: src/controllers/productController.ts [hunk 3]
```diff
@@ -219,13 +219,65 @@ export const getAProduct = async (req: Request, res: Response) => {
...
+            as: 'seller',
+            attributes: ['id', 'name', 'phone', 'email'],
+          },
+          {
+            model: Database.Category,
+            as: 'category',
+            attributes: ['id', 'name'],
+          },
+        ],
+        limit: 4,
```
Comment (fab-ryan) [inline src/controllers/productController.ts]: and also here this is not related

**Review by fab-ryan — CHANGES_REQUESTED** [review]
You need to come up your own algorithm for resolving

File: src/controllers/productController.ts [hunk 4]
```diff
@@ -219,13 +219,41 @@ export const getAProduct = async (req: Request, res: Response) => {
...
-      res,
-      200,
+    const relatedProductsVisitedBySeller = await excOperation<
+      ProductAttributes[]
+    >('Product', 'findAll', {
+      where: {
+        sellerId: product.sellerId,
+      },
+      limit: 4,
+    });
```
Comment (fab-ryan) [inline src/controllers/productController.ts]: no algorithm there

**Review by fab-ryan — CHANGES_REQUESTED** [review]
you didn't provide algorithm for related

**Review by fab-ryan — APPROVED** [review]

**Review by musabehonore — APPROVED** [review]

**Review by RWEMAREMY — APPROVED** [review]
good job

**Review by manzidavidrw — APPROVED** [review]

**Review by Mukunzijames — APPROVED** [review]

**Pull request merged**